### PR TITLE
Added multi-threading to distance matrix calculation

### PR DESCRIPTION
**Pull request by SirReginald1**
Added multi-threading to distance matrix calculation to speed up distance calculation for long texts

### CODE REVIEW

File: tests/test_stemma_NJ.py [hunk 1]
```diff
@@ -77,8 +77,12 @@ def test_getters(self):
         """Testing getters for class properties."""
         self.stemmaNJ = StemmaNJ(distance=levenshtein)
         self.stemmaNJ.compute(folder_path=self.stemma_folder_path)
-        self.assertTrue((self.stemmaNJ.dist_matrix.round(7) == self.test_distance_matrix.round(
-            7)).all(), msg="The distance matrix is not correct.")
+        print(self.stemmaNJ.dist_matrix.round(7))
```
Comment (SphRbtHyk) [inline tests/test_stemma_NJ.py:80]: Print can be removed :-)

File: src/stemmabench/algorithms/stemma_NJ.py [hunk 2]
```diff
@@ -104,13 +106,20 @@ def dist(self, distance: Callable) -> None:
...
+        keys.sort()
+        labels = {}
+        man_range =  range(len(self._manuscripts))
+        with ThreadPoolExecutor(max_workers=len(self._manuscripts)*len(self._manuscripts)) as executor:
+            for col in man_range:
+                for row in man_range:
+                    labels.update({f"{keys[row]},{keys[col]}" : executor.submit(distance, self._manuscripts[keys[row]], self._manuscripts[keys[col]])})
+        for label in labels:
+            labels[label] = labels[label].result()
+        for col, col_num in zip(keys, man_range):
```
Comment (SphRbtHyk) [inline src/stemmabench/algorithms/stemma_NJ.py:120]: can be an enumerate

File: tests/test_stemma_NJ.py (hunk 1 above)
Comment (SirReginald1) [reply tests/test_stemma_NJ.py:80]: Removed

File: src/stemmabench/algorithms/stemma_NJ.py (hunk 2 above)
Comment (SirReginald1) [reply src/stemmabench/algorithms/stemma_NJ.py]: Done

**Pull request merged**